navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/NavigationEngine.java

```diff
@@ -77,19 +77,25 @@ void onLocationChanged(DirectionsRoute directionsRoute, Location location) {
 
     previousLocation = location;
 
-    // Get the new alert level
-    AlertLevelState alertLevelState
-      = new AlertLevelState(location, previousRouteProgress, stepIndex, legIndex, options);
-    int alertLevel = alertLevelState.getNewAlertLevel();
-    stepIndex = alertLevelState.getStepIndex();
-    legIndex = alertLevelState.getLegIndex();
+    if (bearingMatchesManeuverFinalHeading(
+      location, previousRouteProgress, options.getMaxTurnCompletionOffset())) {
+      increaseIndex(previousRouteProgress);
+    }
 
     SnapLocation snapLocation = new SnapLocation(location,
       previousRouteProgress.getCurrentLegProgress().getCurrentStep(), options);
 
     // Create a RouteProgress.create object using the latest user location
     RouteProgress routeProgress = RouteProgress.create(directionsRoute, snapLocation.getUsersCurrentSnappedPosition(),
-      legIndex, stepIndex, alertLevel);
+      legIndex, stepIndex);
+
+    for (Milestone milestone : milestones) {
+      if (milestone.validate(previousRouteProgress, routeProgress)) {
+        for (MilestoneEventListener listener : milestoneEventListeners) {
+          listener.onMilestoneEvent(routeProgress, "instruction", milestone.getIdentifier());
```

Comment:
👍  this is really just a placeholder until I add text instruction logic in a separate PR.